src/main/java/net/neoforged/neoforge/flag/FlagCommand.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.flag;
+
+import static com.mojang.brigadier.Command.SINGLE_SUCCESS;
+import static net.minecraft.commands.Commands.argument;
+import static net.minecraft.commands.Commands.literal;
+
+import com.google.common.collect.Sets;
+import com.mojang.brigadier.builder.ArgumentBuilder;
+import com.mojang.brigadier.builder.LiteralArgumentBuilder;
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.suggestion.Suggestions;
+import com.mojang.brigadier.suggestion.SuggestionsBuilder;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.commands.SharedSuggestionProvider;
+import net.minecraft.commands.arguments.ResourceLocationArgument;
+import net.minecraft.network.chat.ClickEvent;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.HoverEvent;
+import org.jetbrains.annotations.ApiStatus;
+
+@ApiStatus.Internal
+public interface FlagCommand {
+    static ArgumentBuilder<CommandSourceStack, ?> register() {
+        return literal("flag")
+                .executes(FlagCommand::listCommands)
+                .then(stateSubCommand(true))
+                .then(stateSubCommand(false));
+    }
```

Comment:
Suggestion for the command structure here:
```
/neoforge flags
/neoforge flags list
/neoforge flags enabled
/neoforge flags disabled
/neoforge flags set flag enabled
/neoforge flags set flag disabled
/neoforge flags get flag
```

Reply:
Command has been updated to match the above structure, which I will say I do prefer and matches structure of vanilla commands